### PR TITLE
Reenable disabled smoke and integration tests

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR fixes #[20885](https://github.com/microsoft/azuredatastudio/issues/20885)

The PR reenables smoke and integration tests in the build pipeline

### CODE REVIEW

**Review by kburtram — APPROVED** [review]
Thanks for fixing!

File: test/smoke/src/utils.ts [hunk 1]
```diff
@@ -130,15 +131,16 @@ export function getRandomUserDataDir(options: ApplicationOptions): string {
 
 export function installCommonAfterHandlers(opts: minimist.ParsedArgs, appFn?: () => Application | undefined, joinFn?: () => Promise<unknown>) {
 	afterEach(async function () {
-		const app: Application = appFn?.() ?? this.app;
+		// {{SQL CARBON EDIT}} - No clear way to take a screenshot
```
Comment (Charles-Gagnon) [inline test/smoke/src/utils.ts:134]: Do we have an issue tracking re-enabling this? Smoke test failures can be tricky to debug without screenshots so having those would be very helpful.

File: test/smoke/src/sql/areas/notebook/notebook.test.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export function setup(opts: minimist.ParsedArgs) {
 		beforeSuite(opts);
 		afterSuite(opts);
 
-		it('can perform basic text cell functionality', async function () {
+		it.skip('can perform basic text cell functionality', async function () {
```
Comment (Charles-Gagnon) [inline test/smoke/src/sql/areas/notebook/notebook.test.ts:15]: @lewis-sanchez Do we have an issue created to track fixing/re-enabling these skipped tests?

File: test/smoke/src/utils.ts [hunk 3]
```diff
@@ -159,6 +161,61 @@ export function installCommonAfterHandlers(opts: minimist.ParsedArgs, appFn?: ()
...
+
+// {{SQL CARBON EDIT}} - defined afterSuite
+export function afterSuite(opts: minimist.ParsedArgs) {
+	afterEach(async function () {
+		const app = this.app as Application;
+
+		const name = this.currentTest!.fullTitle().replace(/[^a-z0-9\-]/ig, '_');
+
+		if (this.currentTest?.state === 'failed' && opts.screenshots) {
+			await app.code.driver.stopTracing(name, true); // True, captures screenshots.
```
Comment (Charles-Gagnon) [inline test/smoke/src/utils.ts:203]: This seems to conflict with your comment above about not being able to capture screenshots. Is this not working or something?

File: test/smoke/src/main.ts [hunk 4]
```diff
@@ -417,25 +425,26 @@ function createOptions(): ApplicationOptions {
...
+	// Always log to log file
+	loggers.push(new FileLogger(path.join(logsRootPath, 'smoke-test-runner.log')));
 
 	return {
 		quality,
 		codePath: opts.build,
 		workspacePath,
 		userDataDir,
-		extensionsPath,
+		extensionsPath: extensionsPath ?? '', // {{SQL CARBON EDIT}} null coalescing
```
Comment (Charles-Gagnon) [inline test/smoke/src/main.ts:440]: Please make sure any carbon edit comments explain why this is necessary - we can already see that we're null coalescing the value but why this is necessary isn't clear.

File: test/smoke/src/sql/areas/notebook/notebook.test.ts (hunk 2 above)
Comment (lewis-sanchez) [reply test/smoke/src/sql/areas/notebook/notebook.test.ts:15]: I made an issue: https://github.com/microsoft/azuredatastudio/issues/24001

File: test/smoke/src/utils.ts (hunk 3 above)
Comment (lewis-sanchez) [reply test/smoke/src/utils.ts:203]: That comment is from code that was carried over when I was reimplementing the missing methods.

File: test/smoke/src/main.ts (hunk 4 above)
Comment (lewis-sanchez) [reply test/smoke/src/main.ts:440]: I was having a TypeScript typing issue. The issue was: type 'string | undefined' cannot be assigned to type 'string'. Adding null coalescing was how that was resolved.

File: test/smoke/src/utils.ts (hunk 1 above)
Comment (lewis-sanchez) [reply test/smoke/src/utils.ts:134]: Issue is here: https://github.com/microsoft/azuredatastudio/issues/24002

**Pull request merged**